src/core/control/xojfile/LoadHandler.cpp

```diff
@@ -925,6 +926,71 @@ void LoadHandler::parserEndElement(GMarkupParseContext* context, const gchar* el
     }
 }
 
+void LoadHandler::fixNullPressureValues() {
+    /*
+     * Due to various bugs (see e.g. https://github.com/xournalpp/xournalpp/issues/3643), old files may contain strokes
+     * with non-positive pressure values.
+     *
+     * Those strokes thus fails the somewhat reasonable assumption that pressure values should be positive.
+     * The portions of stroke with non-positive pressure values are in any case invisible.
+     *
+     * This function fixes corrupted strokes by
+     *  1- removing every point with a non-positive pressure value.
+     *  2- if required, splitting the affected stroke into several strokes.
+     *  3- entirely deleting strokes without any valid pressure value
+     */
+    auto& pts = stroke->getPointVector();
+    if (pressureBuffer.size() >= pts.size()) {
+        // Too many pressure values. Drop the last ones
+        assert(pts.size() >= 2);  // An error has already been returned otherwise
+        pressureBuffer.resize(pts.size() - 1);
+    }
+
+    auto nextPositive = std::find_if(pressureBuffer.begin(), pressureBuffer.end(), [](double v) { return v > 0; });
+
+    std::vector<std::vector<Point>> strokePortions;
+    while (nextPositive != pressureBuffer.end()) {
+        auto nextNonPositive = std::find_if(nextPositive, pressureBuffer.end(), [](double v) { return v <= 0; });
+        size_t nValidPressureValues = static_cast<size_t>(std::distance(nextPositive, nextNonPositive));
+
+        std::vector<Point> ps;
+        ps.reserve(nValidPressureValues + 1);
+
+        std::transform(nextPositive, nextNonPositive,
+                       std::next(pts.begin(), std::distance(pressureBuffer.begin(), nextPositive)),
+                       std::back_inserter(ps), [](double v, const Point& p) { return Point(p.x, p.y, v); });
+
+        ps.emplace_back(*(pts.begin() + std::distance(pressureBuffer.begin(), nextNonPositive)));
+
+        assert(ps.size() == nValidPressureValues + 1);
+        strokePortions.emplace_back(std::move(ps));
+
+        if (nextNonPositive == pressureBuffer.end()) {
+            break;
+        }
+        nextPositive = std::find_if(nextNonPositive, pressureBuffer.end(), [](double v) { return v > 0; });
+    }
+
+    if (strokePortions.empty()) {
+        // There was no valid pressure values! Delete the stroke entirely
+        g_warning("Found a stroke with only non-positive pressure values! Removing this invisible stroke.");
+        layer->removeElement(stroke, true);
+        stroke = nullptr;
+        return;
+    }
+
+    g_warning("Found a stroke with some non-positive pressure values. Removing the affected points.");
+    for_first_then_each(
+            strokePortions, [&](auto& points) { stroke->setPointVector(points); },
+            [&](auto& points) {
+                Stroke* s = new Stroke();
+                s->applyStyleFrom(stroke);
+                s->setPointVector(points);
```

Comment:
note: it seems like this will make a copy. We might want to define an overload of `setPointVector` that takes an rvalue reference.